src/app/pages/datasets/components/dataset-form/dataset-form.component.ts

```diff
@@ -111,22 +111,22 @@ export class DatasetFormComponent implements OnInit {
     this.isLoading = true;
     this.cdr.markForCheck();
 
-    this.datasetFormService.ensurePathLimits(this.slideInData.datasetId).pipe(
+    this.datasetFormService.isPathLengthAndDepthSafe(this.slideInData.datasetId).pipe(
```

Comment:
Method name now implies that there will be no side effects like closing a slide-in. You either need to use a different name or move side-effect outside of the method.